### DISCUSSION TITLE
Can I print out .text section of a binary from ghidra using headless or ghidrascript?? if so, how would i do it?

### DISCUSSION THREAD

**Discussion by anzosasuke** [NationalSecurityAgency/ghidra#3041] [General]
Can I print out .text section of a binary from ghidra?? if so, how would i do it?

**Comment by dragonmacher** [comment]
You should be able to:

1. select the .text section (use Select -> Bytes...)
2. export the selected contents (File -> Export)

Choose ASCII or HTML for the output.  Be sure to check the box that limits the output to your selection.   There is an options button on the export dialog that allows you to configure what gets printed.

You could also perform more fine-grained output via a custom `GhidraScript`.

**Reply by anzosasuke** [reply]
Thanks for replying. I want to do it from ghidraScript or headless method. But I don't know how. Could you help me with this?

**Reply by astrelsky** [reply]
> Thanks for replying. I want to do it from ghidraScript or headless method. But I don't know how. Could you help me with this?

I recommend keeping this page handy [FlatProgramAPI](https://ghidra.re/ghidra_docs/api/ghidra/program/flatapi/FlatProgramAPI.html). See the `getMemoryBlock` method.

Side note, a python script has all these methods available to it, you do not need an instance of this class.

**Reply by anzosasuke** [reply]
I want to feed these data to a machine learning model. thanks so much

**Reply by astrelsky** [reply]
> I want to feed these data to a machine learning model. thanks so much

Hmm. I'm not sure of the specifics of what you're doing, but have you considered feeding it the pcode?

**Reply by anzosasuke** [reply]
I think I could do that in future. but I am just starting.

**Reply by anzosasuke** [reply]
while trying to  access getMemoryBlock('.text'), I am getting this error. I am a newbie and using jython to exploit API is kinda new thing for me. Can you help me with this?

AttributeError: 'ghidra.program.database.ProgramDB' object has no attribute 'getMemoryBlock'

**Reply by astrelsky** [reply]
Change `currentProgram.getMemoryBlock('.text')` to `getMemoryBlock('.text')`.

**Reply by anzosasuke** [reply]
It seems to give me memory block DB@ some address. I tried getting the bytes inside of .text section. It's not iterable. memoryblockInputStream is not a string or buffer to write in a file. I can't seem to find a way to get those bytes inside .text section. I must be dumb. Can you please ,may be help me with this one too.?

**Reply by anzosasuke** [reply]
offset = 1052832
size = 533
data_in = getMemoryBlock('.text').getData()
data = data_in.read(byte[] b, 1052832, 533)  # I am getting error here. tried using bytes instead of byte[] b, I am doing somethinf wrong here, I dont know what that is ?
